api/environments/component_handler.go

```diff
@@ -126,6 +131,64 @@ func (eh EnvironmentHandler) RestartComponent(appName, envName, componentName st
 	return nil
 }
 
+// RestartComponentAuxiliaryResource Restarts a component's auxiliary resource
+func (eh EnvironmentHandler) RestartComponentAuxiliaryResource(appName, envName, componentName, auxType string) error {
+	log.Infof("Restarting auxiliary resource %s for component %s, %s", auxType, componentName, appName)
+
+	deploySummary, err := eh.deployHandler.GetLatestDeploymentForApplicationEnvironment(appName, envName)
```

Comment:
This call created deployment with components and jobs (using `deploy.getDeployments(namespace, appName, "", latest)`, with `latest: true`), while just only last (active) deployment name is needed, to repeat the loop in the next operator `eh.deployHandler.GetDeploymentWithName(appName, deploySummary.Name)`, which also use `deploy.getDeployments(namespace, appName, "", latest)`, but with `latest: false`. 
Maybe we can reduce this complexity?

Reply:
Changed to use another method